test/unit/spec/EnrollCall_spec.js

```diff
@@ -44,37 +41,38 @@ function (Q, Okta, OktaAuth, LoginUtil, Util, Form, Beacon, Expect, $sandbox,
       router.on('afterError', afterRenderHandler);
       Util.registerRouter(router);
       Util.mockRouterNavigate(router, startRouter);
-      return Q()
-        .then(function () {
-          setNextResponse(resp || resAllFactors);
-          return Util.mockIntrospectResponse(router, resp || resAllFactors);
-        })
-        .then(function () {
-          setNextResponse(resp || resAllFactors);
-          router.refreshAuthState('dummy-token');
-          return Expect.waitForEnrollChoices();
-        })
-        .then(function () {
-          router.enrollCall();
-          return Expect.waitForEnrollCall({
-            router: router,
-            beacon: new Beacon($sandbox),
-            form: new Form($sandbox),
-            ac: authClient,
-            setNextResponse: setNextResponse,
-            afterRenderHandler: afterRenderHandler
+
+      var test = {
+        router: router,
+        beacon: new Beacon($sandbox),
+        form: new Form($sandbox),
+        ac: authClient,
+        setNextResponse: setNextResponse,
+        afterRenderHandler: afterRenderHandler
+      };
+
+      const enrollCall = (test) => {
+        setNextResponse(resp || resAllFactors);
+        router.refreshAuthState('dummy-token');
+        return Expect.waitForEnrollChoices(test)
+          .then(function (test) {
+            test.router.enrollCall();
+            return Expect.waitForEnrollCall(test);
+          })
+          .then(test => {
+            spyOn(test.router.controller, 'trapAuthResponse').and.callThrough();
+            return test;
           });
-        })
-        .then(test => {
-          spyOn(test.router.controller, 'trapAuthResponse').and.callThrough();
-          return test;
-        });
-    }
+      };
 
-    function setupWithFactorEnroll () {
-      return setup(resAllFactorsIdx);
+      if (startRouter) {
+        return Expect.waitForPrimaryAuth(test).then(enrollCall);
```

Comment:
yeah, after debugging through the code we found that if startRouter is set to true it will try to bootstrap the widget and load primary auth first, hence we have a wait for that. When we had `tick` or `Q` before it was doing the same wait but with this we exactly know what we are waiting for and we only wait when startRouter is true.